evap/results/templates/results_evaluation_detail.html

```diff
@@ -73,12 +73,12 @@ <h3>{{ evaluation.full_name }} ({{ evaluation.course.semester.name }})</h3>
         <div class="card-header d-flex">
             <div class="mr-auto">{% trans 'Overview' %}</div>
             {% if view == "export" or view == "full" %}
-                <a href="{% url 'results:evaluation_text_answers_export' evaluation.id %}?view={{ view }}{% if contributor_id is not None %}&contributor_id={{ contributor_id }}{% endif %}" type="button" class="btn btn-sm btn-light d-print-none">
+                <a class="btn btn-sm btn-light d-print-none" href="{% url 'results:evaluation_text_answers_export' evaluation.id %}?view={{ view }}{% if contributor_id is not None %}&contributor_id={{ contributor_id }}{% endif %}" type="button">
                     {% trans 'Export text answers' %}
                 </a>
             {% endif %}
             {% if evaluation.course.grade_documents.count == 1 and can_download_grades %}
-                <a class="btn btn-sm btn-light d-print-none" href="{% url 'grades:download_grades' evaluation.course.grade_documents.first.id %}">{{ evaluation.course.grade_documents.first.description }}</a>
+                <a class="btn btn-sm btn-light d-print-none ml-2" href="{% url 'grades:download_grades' evaluation.course.grade_documents.first.id %}">{{ evaluation.course.grade_documents.first.description }}</a>
             {% elif evaluation.course.grade_documents.count > 1 and can_download_grades %}
                 <div class="btn-group d-print-none" role="group">
                     <button type="button" id="btnDownload" role="button" class="btn btn-sm btn-light dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">{% trans 'Grades' %}</button>
```

Comment:
you should add the margin for this element as well in case there are multiple documents available

Reply:
Right, I did not read this properly.

Reply:
`57fa7f0983fe9ff31253aab286fdc6d3a25d5dfc` onward:

I added a class `InputData` to the exporter that makes the exporting a bit more readable, since it got really really awful after fixing the first error.
I had to handcraft an instance of `InputData` in the tests (there is a comment that explains why), should I add a test for the `InputData.__init__` method?
Or maybe this is not the optimal approach anyway and I should change it again? Let me know what you think?

Again, I can squash all these fixups after we settled everything.